implementations/rust/ockam/ockam_command/src/credential/store.rs

```diff
@@ -74,14 +76,37 @@ async fn run_impl(
             }
         };
 
-        // store
-        opts.state.credentials.create(
-            &cmd.credential_name,
-            CredentialConfig::new(cmd.identity().await?, cred_as_str.to_string())?,
-        )?;
+        let vault_name = cmd
+            .vault
+            .clone()
+            .unwrap_or_else(|| default_vault_name(&opts.state));
 
-        *is_finished.lock().await = true;
-        Ok(cred_as_str)
+        let issuer = match &cmd.identity().await {
+            Ok(i) => i,
+            Err(_) => {
+                *is_finished.lock().await = true;
+                return Ok((false, "Issuer is invalid".to_string()));
+            }
+        }
+        .identifier();
+
+        match validate_encoded_cred(&cred_as_str, &issuer, &vault_name, &opts).await
+        {
+            Ok(_) => {
+                // store
+                opts.state.credentials.create(
+                    &cmd.credential_name,
+                    CredentialConfig::new(cmd.identity().await?, cred_as_str.clone())?,
+                )?;
+
+                *is_finished.lock().await = true;
+                Ok((true, cred_as_str))
+            },
+            Err(e) => {
+                *is_finished.lock().await = true;
+                Ok((false, format!("Credential is invalid\n{}", e.to_string())))
+            }
+        }
```

Comment:
There are some simplifications you can do here. Note that you are returning a tuple with an ok/err pair. You could change that and return a plain Result<String> with the credential.

```rust
let issuer = match &cmd.identity().await {
    Ok(i) => i,
    Err(_) => {
        *is_finished.lock().await = true;
        return Err(miette!("Issuer is invalid").into());
    }
}
.identifier();
if validate_encoded_cred(&cred_as_str, &issuer, &vault_name, &opts)
    .await
    .is_err()
{
    *is_finished.lock().await = true;
    return Err(miette!("Credential is invalid").into());
};
Ok(cred_as_str)
```